tdvt/tdvt/config_gen/datasource_list.py

```diff
@@ -358,27 +357,55 @@ def load_test(config, test_dir=get_root_dir()):
 class TestRegistry(object):
     """Add a new datasource here and then add it to the appropriate registries below."""
 
-    def __init__(self, ini_file):
+    def __init__(self, ini_file, ds):
         self.dsnames = {}
         self.suite_map = {}
+        self.ds = ds
+
+        # When the ds is a suite, we assign the datasources in the suite to the ds variable.
+        registry_ini_file = get_ini_path_local_first('config/registry', ini_file)
+        self.load_registry(registry_ini_file)
+        self.create_ds_list()
+
+        # Check if any list items are suites themselves. If so we include the datasources in those as well.
+        suites_list = ['all', 'standard', 'all_passing', 'nightly', 'slow', 'all_bigquery_sql', 'all_bigquery', 'all_redshift', 'perf']
+        for x in self.ds:
+            if x in suites_list:
+                self.create_ds_list()
+
+        print(self.ds)
 
         # Read all the datasource ini files and load the test configuration.
         ini_files = get_all_ini_files_local_first('config')
-        for f in ini_files:
-            logging.debug("Reading ini file [{}]".format(f))
-            config = configparser.ConfigParser()
-            # Preserve the case of elements.
-            config.optionxform = str
-            try:
-                config.read(f)
-            except configparser.ParsingError as e:
-                logging.debug(e)
-                continue
-
-            self.add_test(load_test(config))
 
+        for d in self.ds:
+            # Include only the ini files that apply to a datasource in the list.
+            for f in ini_files:
+                if d == os.path.basename(f).replace(".ini", ""):
```

Comment:
```suggestion
                if d == Path(f).stem:
```

Reply:
`PurePath`/`Path` has a bunch of [really useful methods](https://docs.python.org/3/library/pathlib.html#pathlib.PurePath.stem)